### ISSUE TITLE
into-ing an empty table from postgres to bcolz break with StopIteration

### ISSUE THREAD

**Issue by teh** [ContinuumIO/odo#205]
With a fairly recent version

```
>>> odo.__version__
'2f05cf90d1fcc5d95eaa55682f08b9d1b06f87bb'
```

I get:

```
Traceback (most recent call last):
  File "get_data.py", line 66, in <module>
    main()
  File "get_data.py", line 61, in main
    odo.odo(db_url.format(x), out_path)
  File "/nix/store/4kjkmcfmblg6wfriwpadbcjypq8ncrrb-python2.7-odo-0.3.1/lib/python2.7/site-packages/odo/odo.py", line 90, in odo
    return into(target, source, **kwargs)
  File "/nix/store/4rsj0f8kcsmxh66gcq4sydahcjqpmdjk-python2.7-multipledispatch-0.4.7/lib/python2.7/site-packages/multipledispatch/dispatcher.py", line 163, in __call__
    return func(*args, **kwargs)
  File "/nix/store/4kjkmcfmblg6wfriwpadbcjypq8ncrrb-python2.7-odo-0.3.1/lib/python2.7/site-packages/odo/into.py", line 120, in into_string_string
    return into(a, r, **kwargs)
  File "/nix/store/4rsj0f8kcsmxh66gcq4sydahcjqpmdjk-python2.7-multipledispatch-0.4.7/lib/python2.7/site-packages/multipledispatch/dispatcher.py", line 163, in __call__
    return func(*args, **kwargs)
  File "/nix/store/4kjkmcfmblg6wfriwpadbcjypq8ncrrb-python2.7-odo-0.3.1/lib/python2.7/site-packages/odo/into.py", line 114, in into_string
    return into(a, b, dshape=ds, **kwargs)
  File "/nix/store/4rsj0f8kcsmxh66gcq4sydahcjqpmdjk-python2.7-multipledispatch-0.4.7/lib/python2.7/site-packages/multipledispatch/dispatcher.py", line 163, in __call__
    return func(*args, **kwargs)
  File "/nix/store/4kjkmcfmblg6wfriwpadbcjypq8ncrrb-python2.7-odo-0.3.1/lib/python2.7/site-packages/odo/into.py", line 100, in into_object
    return append(target, source, **kwargs)
  File "/nix/store/4rsj0f8kcsmxh66gcq4sydahcjqpmdjk-python2.7-multipledispatch-0.4.7/lib/python2.7/site-packages/multipledispatch/dispatcher.py", line 163, in __call__
    return func(*args, **kwargs)
  File "/nix/store/4kjkmcfmblg6wfriwpadbcjypq8ncrrb-python2.7-odo-0.3.1/lib/python2.7/site-packages/odo/backends/bcolz.py", line 34, in numpy_append_to_bcolz
    return append(a, convert(chunks(np.ndarray), b, **kwargs), **kwargs)
  File "/nix/store/4kjkmcfmblg6wfriwpadbcjypq8ncrrb-python2.7-odo-0.3.1/lib/python2.7/site-packages/odo/core.py", line 30, in __call__
    return _transform(self.graph, *args, **kwargs)
  File "/nix/store/4kjkmcfmblg6wfriwpadbcjypq8ncrrb-python2.7-odo-0.3.1/lib/python2.7/site-packages/odo/core.py", line 46, in _transform
    x = f(x, excluded_edges=excluded_edges, **kwargs)
  File "/nix/store/4kjkmcfmblg6wfriwpadbcjypq8ncrrb-python2.7-odo-0.3.1/lib/python2.7/site-packages/odo/convert.py", line 192, in iterator_to_numpy_chunks
    first, rest = next(seq2), seq2
  File "/nix/store/hn8pdkg1rdy520hvscsvcy558cyhpdfy-python2.7-toolz-0.7.1/lib/python2.7/site-packages/toolz/itertoolz.py", line 669, in partition_all
    prev = next(it)
StopIteration
```

The table is just empty (0 rows).

**Comment by cpcloud** [comment]
What are you trying to do here? It seems strange to convert an empty table to another format. The proper way to do this would be to create the table from the inferred dshape of the postgresql table:

```python
In [3]: bc = resource('f.bcolz', dshape=discover('postgresql://localhost::foo'))

In [4]: bc
Out[4]:
ctable((0,), [('a', '<i4'), ('b', 'S30')])
  nbytes: 0; cbytes: 32.00 KB; ratio: 0.00
  cparams := cparams(clevel=5, shuffle=True, cname='blosclz')
  rootdir := 'f.bcolz'
[]
```

**Comment by cpcloud** [comment]
@teh on a second look, this was a bug and is fixed in #255
